lms/djangoapps/instructor/views/instructor_dashboard.py

```diff
@@ -65,8 +66,33 @@ def is_enabled(cls, course, user=None):  # pylint: disable=unused-argument,redef
         return user and has_access(user, 'staff', course, course.id)
 
 
+def check_staff_or_404():
```

Comment:
@dan-f There is no check like that in old instructor dashboard. I just refactored this due to a unit test failure.